tests/end_to_end/end_to_end.cpp

```diff
@@ -1227,6 +1228,42 @@ _xdp_encap_reflect_packet_test(ebpf_execution_type_t execution_type, ADDRESS_FAM
     }
 }
 
+TEST_CASE("printk", "[end_to_end]")
+{
+    _test_helper_end_to_end test_helper;
+    single_instance_hook_t hook(EBPF_PROGRAM_TYPE_BIND, EBPF_ATTACH_TYPE_BIND);
+    program_info_provider_t bind_program_info(EBPF_PROGRAM_TYPE_BIND);
+    uint32_t ifindex = 0;
+    program_load_attach_helper_t program_helper(
+        SAMPLE_PATH "printk.o",
+        EBPF_PROGRAM_TYPE_BIND,
```

Comment:
suggestion: consider using sample extension for such test programs that does not really need any networking features for testing.

Reply:
I wanted a real program that would actually do something when installed so I could test it in reality and it could actually be a realistic sample.